### DISCUSSION TITLE
ViewSettingsDialog event detail outcome

### DISCUSSION THREAD

**Discussion by FRicardi** [UI5/webcomponents#6907] [Q&A]
When I was studying about ViewSettingsDialog, I was wondering how to deal with the event details object. I see that it uses the `text` property to populate the object's keys, which can be troublesome on cases that uses i18n to display the texts. 

I believe that there's a workaround that involves using the translations as key maps, but I wonder if there's an easier way to manipulate filter and sort keys to avoid these kinds of workarounds.

Have someone stumbled upon this problem as well? What are the possible solutions?

https://codesandbox.io/s/table-filter-sort-and-grouping-7w389k?file=/src/App.js

**Comment by Lukas742** [comment]
Hi @FRicardi 

The component offers a way to get the current sort item (`sortByItem`) in the events, you can apply e.g. data-attributes (as described [here](https://sap.github.io/ui5-webcomponents-react/?path=/docs/knowledge-base-ids-via-dataset--docs)) to assign different values than the text and for the order you can use the `sortDescending ` boolean. 

However I don't see such options for the filters, and it seems like that the documentation is not correct there. In the documentation it says the events offer a `filterItems` property, but it's called `filters`. Also only the text attributes are used there and not the actual elements, what makes it impossible getting other values.

Since the `ViewSettingsDialog` and its sub-components are web components, developed by the ui5-webcomponents team, I'll forward this discussion to their repo. Since this could also be considered a feature request (maybe even a bug, depending on perspective), you can then decide if you want to turn this discussion into an issue.

___

Hi colleagues,

could you please help @FRicardi with this issue?

Here you can find a codeSandbox using plain js with UI5WC:
https://codesandbox.io/s/ui5-webcomponents-forked-bg03lg?file=/src/index.js

**Reply by FRicardi** [reply]
It would be nice to have that feature for easier maintenance  of codebase indeed, therefore I will create the issue from this discussion.
Thanks for the quick response, have a nice week!

**Comment by DMihaylova** [comment]
Closed via https://github.com/SAP/ui5-webcomponents/pull/9838
